src/vcBPA.cpp

```diff
@@ -1067,6 +1067,9 @@ void vcBPA_CompareExport(vcState *pProgramState, vdkPointCloud *pOldModel, vdkPo
   vdkConvertCustomItem item = {};
   item.pName = "DisplacementComparison";
 
+  if (SRID != -1)
+    item.srid = SRID;
```

Comment:
Couldn't this be pulled out of `pNewModel` instead of being passed in, we already have the `pNewModel` and we don't want the SRID to be overridden by anything ever.

Reply:
it could, as in vcModel_LoadMetadata, but it is a lot of work that's already done when loading the model
`if (vdkPointCloud_GetMetadata(pModel->m_pPointCloud, &pMetadata) == vE_Success)
  {
    pModel->m_metadata.Parse(pMetadata);
    pModel->m_hasWatermark = pModel->m_metadata.Get("Watermark").IsString();

    pModel->m_meterScale = pModel->m_pointCloudHeader.unitMeterScale;
    pModel->m_pivot = udDouble3::create(pModel->m_pointCloudHeader.pivot[0], pModel->m_pointCloudHeader.pivot[1], pModel->m_pointCloudHeader.pivot[2]);

    vcSRID srid = 0;
    const char *pSRID = pModel->m_metadata.Get("ProjectionID").AsString();
    const char *pWKT = pModel->m_metadata.Get("ProjectionWKT").AsString();

    if (pSRID != nullptr)
    {
      pSRID = udStrchr(pSRID, ":");
      if (pSRID != nullptr)
        srid = udStrAtou(&pSRID[1]);

      pModel->m_pPreferredProjection = udAllocType(udGeoZone, 1, udAF_Zero);
      if (udGeoZone_SetFromSRID(pModel->m_pPreferredProjection, srid) != udR_Success)
        udFree(pModel->m_pPreferredProjection);
    }

    if (pModel->m_pPreferredProjection == nullptr && pWKT != nullptr)
    {
      pModel->m_pPreferredProjection = udAllocType(udGeoZone, 1, udAF_Zero);

      if (udGeoZone_SetFromWKT(pModel->m_pPreferredProjection, pWKT) != udR_Success)
        udFree(pModel->m_pPreferredProjection);
    }...`

Reply:
Would you prefer I do that in vcBPA_CompareExport?

Reply:
You need to use three backticks when posting multiple lines of code, you can also add `c++` after the first three backticks and it will format it as C++ code.

You might be able to forgo the SRID variable entirely and just feed all of the metadata back into the conversion system with `vdkConvert_SetMetadata` for each key value pair in the JSON object. Additionally, if the SRID is defined through the WKT, then we should use the WKT not the SRID as that's what the original model did too.